clang/lib/Lex/PPMacroExpansion.cpp

```diff
@@ -1604,10 +1604,12 @@ static bool isTargetVariantEnvironment(const TargetInfo &TI,
   return false;
 }
 
-#if defined(__sun__) && defined(__svr4__)
+#if defined(__sun__) && defined(__svr4__) && defined(__clang__) &&             \
+    __clang__ < 20
```

Comment:
Does every version of gcc we support handles that?

Reply:
Certainly: the fix for GCC PR libstdc++-v3/1773 went in 13 years ago.  That should be good enough ;-)